In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('C:\\Users\\Vighnesh\\Desktop\\PowerBI Files\\Diwali Sales Data.csv', encoding = 'unicode_escape')

#import csv file and store in variable df

In [3]:
df.head(10)   #Display first ten rows of the datatset

User_ID  Cust_name Product_ID Gender Age Group  Age  Marital_Status  \
0  1002903  Sanskriti  P00125942      F     26-35   28               0   
1  1000732     Kartik  P00110942      F     26-35   35               1   
2  1001990      Bindu  P00118542      F     26-35   35               1   
3  1001425     Sudevi  P00237842      M      0-17   16               0   
4  1000588       Joni  P00057942      M     26-35   28               1   
5  1000588       Joni  P00057942      M     26-35   28               1   
6  1001132       Balk  P00018042      F     18-25   25               1   
7  1002092   Shivangi  P00273442      F       55+   61               0   
8  1003224     Kushal  P00205642      M     26-35   35               0   
9  1003650      Ginny  P00031142      F     26-35   26               1   

              State      Zone       Occupation Product_Category  Orders  \
0       Maharashtra   Western       Healthcare             Auto       1   
1    Andhra Pradesh  Southern             Govt             Auto       3   
2     Uttar Pradesh   Central       Automobile             Auto       3   
3         Karnataka  Southern     Construction             Auto       2   
4           Gujarat   Western  Food Processing             Auto       2   
5  Himachal Pradesh  Northern  Food Processing             Auto       1   
6     Uttar Pradesh   Central           Lawyer             Auto       4   
7       Maharashtra   Western        IT Sector             Auto       1   
8     Uttar Pradesh   Central             Govt             Auto       2   
9    Andhra Pradesh  Southern            Media             Auto       4   

     Amount  Status  unnamed1  
0  23952.00     NaN       NaN  
1  23934.00     NaN       NaN  
2  23924.00     NaN       NaN  
3  23912.00     NaN       NaN  
4  23877.00     NaN       NaN  
5  23877.00     NaN       NaN  
6  23841.00     NaN       NaN  
7       NaN     NaN       NaN  
8  23809.00     NaN       NaN  
9  23799.99     NaN       NaN

In [4]:
df.tail(10)   #display last 10 rows of the dataset

User_ID    Cust_name Product_ID Gender Age Group  Age  Marital_Status  \
11241  1003032     Matthias  P00058042      F     26-35   33               0   
11242  1004344   Hildebrand  P00185442      F     26-35   27               1   
11243  1005446      Sheetal  P00297742      M     51-55   53               0   
11244  1005446      Sheetal  P00297742      M     51-55   53               0   
11245  1004140    Bertelson  P00057442      F     26-35   31               1   
11246  1000695      Manning  P00296942      M     18-25   19               1   
11247  1004089  Reichenbach  P00171342      M     26-35   33               0   
11248  1001209        Oshin  P00201342      F     36-45   40               0   
11249  1004023       Noonan  P00059442      M     36-45   37               0   
11250  1002744      Brumley  P00281742      F     18-25   19               0   

                State      Zone   Occupation Product_Category  Orders  Amount  \
11241           Delhi   Central  Hospitality           Office       3   384.0   
11242           Delhi   Central   Healthcare           Office       2   382.0   
11243         Gujarat   Western   Healthcare           Office       1   382.0   
11244  Madhya Pradesh   Central   Healthcare           Office       2   382.0   
11245           Delhi   Central     Aviation           Office       2   381.0   
11246     Maharashtra   Western     Chemical           Office       4   370.0   
11247         Haryana  Northern   Healthcare       Veterinary       3   367.0   
11248  Madhya Pradesh   Central      Textile           Office       4   213.0   
11249       Karnataka  Southern  Agriculture           Office       3   206.0   
11250     Maharashtra   Western   Healthcare           Office       3   188.0   

       Status  unnamed1  
11241     NaN       NaN  
11242     NaN       NaN  
11243     NaN       NaN  
11244     NaN       NaN  
11245     NaN       NaN  
11246     NaN       NaN  
11247     NaN       NaN  
11248     NaN       NaN  
11249     NaN       NaN  
11250     NaN       NaN

In [5]:
df.shape   #Print rows and columns in the dataset

(11251, 15)

In [6]:
df.dtypes    #Print each column and its datatype. 

User_ID               int64
Cust_name            object
Product_ID           object
Gender               object
Age Group            object
Age                   int64
Marital_Status        int64
State                object
Zone                 object
Occupation           object
Product_Category     object
Orders                int64
Amount              float64
Status              float64
unnamed1            float64
dtype: object

<span style = "font-size: 20px;"> It can be seen from df.head() that two columns - Status and Unnamed1, are unnecessary and have null values. These columns serve no purpose and should be dropped </span>

In [7]:
columns = df.columns.tolist()
columns

#stores all columns of the dataset in the form of a list

['User_ID',
 'Cust_name',
 'Product_ID',
 'Gender',
 'Age Group',
 'Age',
 'Marital_Status',
 'State',
 'Zone',
 'Occupation',
 'Product_Category',
 'Orders',
 'Amount',
 'Status',
 'unnamed1']

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11251 entries, 0 to 11250
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   User_ID           11251 non-null  int64  
 1   Cust_name         11251 non-null  object 
 2   Product_ID        11251 non-null  object 
 3   Gender            11251 non-null  object 
 4   Age Group         11251 non-null  object 
 5   Age               11251 non-null  int64  
 6   Marital_Status    11251 non-null  int64  
 7   State             11251 non-null  object 
 8   Zone              11251 non-null  object 
 9   Occupation        11251 non-null  object 
 10  Product_Category  11251 non-null  object 
 11  Orders            11251 non-null  int64  
 12  Amount            11239 non-null  float64
 13  Status            0 non-null      float64
 14  unnamed1          0 non-null      float64
dtypes: float64(3), int64(4), object(8)
memory usage: 1.3+ MB


In [10]:
df.describe()     #Works only on numerical columns 

User_ID           Age  Marital_Status        Orders        Amount  \
count  1.125100e+04  11251.000000    11251.000000  11251.000000  11239.000000   
mean   1.003004e+06     35.421207        0.420318      2.489290   9453.610858   
std    1.716125e+03     12.754122        0.493632      1.115047   5222.355869   
min    1.000001e+06     12.000000        0.000000      1.000000    188.000000   
25%    1.001492e+06     27.000000        0.000000      1.500000   5443.000000   
50%    1.003065e+06     33.000000        0.000000      2.000000   8109.000000   
75%    1.004430e+06     43.000000        1.000000      3.000000  12675.000000   
max    1.006040e+06     92.000000        1.000000      4.000000  23952.000000   

       Status  unnamed1  
count     0.0       0.0  
mean      NaN       NaN  
std       NaN       NaN  
min       NaN       NaN  
25%       NaN       NaN  
50%       NaN       NaN  
75%       NaN       NaN  
max       NaN       NaN

In [10]:
df.isnull().sum()

User_ID                 0
Cust_name               0
Product_ID              0
Gender                  0
Age Group               0
Age                     0
Marital_Status          0
State                   0
Zone                    0
Occupation              0
Product_Category        0
Orders                  0
Amount                 12
Status              11251
unnamed1            11251
dtype: int64

<span style = "font-size: 20px;"> The columns Amount, Status and Unnamed1 have null values. </span>

In [11]:
df[df.Amount.isnull()]

User_ID Cust_name Product_ID Gender Age Group  Age  Marital_Status  \
7    1002092  Shivangi  P00273442      F       55+   61               0   
14   1003858      Cano  P00293742      M     46-50   46               1   
16   1005447       Amy  P00275642      F     46-50   48               1   
109  1005265    Sakshi  P00296242      F     46-50   48               1   
111  1005261   Apoorva  P00057942      F     36-45   41               1   
184  1005538    Kartik  P00269542      F     46-50   49               1   
293  1000326  Jonathan  P00120542      M     51-55   53               0   
344  1002507   Lakshmi  P00045842      F     26-35   35               1   
345  1004498   Srishti  P00030842      F     51-55   55               0   
452  1004601    Gaurav  P00014442      F     36-45   40               1   
464  1004528    Anurag  P00338442      F     26-35   33               1   
493  1002994    Hemant   P0009942      F     36-45   38               0   

              State      Zone   Occupation  Product_Category  Orders  Amount  \
7       Maharashtra   Western    IT Sector              Auto       1     NaN   
14   Madhya Pradesh   Central  Hospitality              Auto       3     NaN   
16   Andhra Pradesh  Southern    IT Sector              Auto       3     NaN   
109           Delhi   Central      Banking  Footwear & Shoes       1     NaN   
111           Delhi   Central    IT Sector  Footwear & Shoes       2     NaN   
184       Karnataka  Southern      Banking  Footwear & Shoes       1     NaN   
293         Gujarat   Western    IT Sector  Footwear & Shoes       3     NaN   
344         Gujarat   Western     Chemical         Furniture       1     NaN   
345           Delhi   Central      Textile  Footwear & Shoes       1     NaN   
452  Madhya Pradesh   Central  Hospitality              Food       4     NaN   
464   Uttar Pradesh   Central   Automobile              Food       2     NaN   
493   Uttar Pradesh   Central    IT Sector              Food       4     NaN   

     Status  unnamed1  
7       NaN       NaN  
14      NaN       NaN  
16      NaN       NaN  
109     NaN       NaN  
111     NaN       NaN  
184     NaN       NaN  
293     NaN       NaN  
344     NaN       NaN  
345     NaN       NaN  
452     NaN       NaN  
464     NaN       NaN  
493     NaN       NaN

In [12]:
#Drop unnecessary columns which are not required.
df.drop(['Status', 'unnamed1'], axis = 1, inplace = True)

In [13]:
df.isnull().sum()

User_ID              0
Cust_name            0
Product_ID           0
Gender               0
Age Group            0
Age                  0
Marital_Status       0
State                0
Zone                 0
Occupation           0
Product_Category     0
Orders               0
Amount              12
dtype: int64

Unnecessary columns have been dropped

In [14]:
#Drop the entire row containing NA or NULL values. inplace = True ensures that the change is complete.
df.dropna(inplace=True)

In [15]:
df.shape

(11239, 13)

Number of rows changed from 11251 to 11239, confirming that 12 rows have been dropped.

In [16]:
df['Amount'] = df['Amount'].astype('int')

#Change the datatype from float to int.

In [18]:
df.rename(columns = {'Cust_name': 'Customer_name', 'User_ID': 'UserID', 'Product_ID': 'ProductID'}, inplace = True)

#Renaming columns for uniformity and readability.

In [19]:
df['Gender'] = df['Gender'].map({'F': 'Female', 'M': 'Male'})

In [20]:
df.drop_duplicates()

UserID Customer_name  ProductID  Gender Age Group  Age  \
0      1002903     Sanskriti  P00125942  Female     26-35   28   
1      1000732        Kartik  P00110942  Female     26-35   35   
2      1001990         Bindu  P00118542  Female     26-35   35   
3      1001425        Sudevi  P00237842    Male      0-17   16   
4      1000588          Joni  P00057942    Male     26-35   28   
...        ...           ...        ...     ...       ...  ...   
11246  1000695       Manning  P00296942    Male     18-25   19   
11247  1004089   Reichenbach  P00171342    Male     26-35   33   
11248  1001209         Oshin  P00201342  Female     36-45   40   
11249  1004023        Noonan  P00059442    Male     36-45   37   
11250  1002744       Brumley  P00281742  Female     18-25   19   

       Marital_Status           State      Zone       Occupation  \
0                   0     Maharashtra   Western       Healthcare   
1                   1  Andhra Pradesh  Southern             Govt   
2                   1   Uttar Pradesh   Central       Automobile   
3                   0       Karnataka  Southern     Construction   
4                   1         Gujarat   Western  Food Processing   
...               ...             ...       ...              ...   
11246               1     Maharashtra   Western         Chemical   
11247               0         Haryana  Northern       Healthcare   
11248               0  Madhya Pradesh   Central          Textile   
11249               0       Karnataka  Southern      Agriculture   
11250               0     Maharashtra   Western       Healthcare   

      Product_Category  Orders  Amount  
0                 Auto       1   23952  
1                 Auto       3   23934  
2                 Auto       3   23924  
3                 Auto       2   23912  
4                 Auto       2   23877  
...                ...     ...     ...  
11246           Office       4     370  
11247       Veterinary       3     367  
11248           Office       4     213  
11249           Office       3     206  
11250           Office       3     188  

[11231 rows x 13 columns]

There are no duplicate rows in the dataset, as the no. of rows remains the same. Dataset has been cleaned and is ready for EDA and further analysis.

In [21]:
df.to_csv('cleaned_diwali_sales.csv', index=False)

In [22]:
import os
print(os.getcwd())

C:\Users\Vighnesh
